man/systemd.exec.xml

```diff
@@ -1778,12 +1778,30 @@ CapabilityBoundingSet=~CAP_B CAP_C</programlisting>
         <varname>RuntimeDirectory=</varname>, one or more directories including their parents by the specified names
-        will be created below <filename>/var/lib</filename>, <filename>/var/cache</filename>, <filename>/var/log</filename>,
-        or <filename>/etc</filename>, respectively, when the unit is started.
-        Unlike <varname>RuntimeDirectory=</varname>, the directories are not removed when the unit is stopped.
-        The lowest subdirectories will be owned by the user and group specified in <varname>User=</varname>
-        and <varname>Group=</varname>. The options imply <varname>ReadWritePaths=</varname>.
-        </para></listitem>
+        will be created below <filename>/var/lib</filename>, <filename>/var/cache</filename>,
+        <filename>/var/log</filename>, or <filename>/etc</filename>, respectively, when the unit is started. Unlike
+        <varname>RuntimeDirectory=</varname>, the directories are not removed when the unit is stopped.</para>
+
+        <para>With the exception of <varname>ConfigurationDirectory=</varname>, the innermost specified directories
+        will be owned by the user and group specified in <varname>User=</varname> and <varname>Group=</varname>. If the
+        specified directories already exist and their owning user or group do not match the configured ones, all files
+        and directories below the specified directories as well as the directories themselves will have their file
+        ownership recursively changed to match what is configured. As an optimization, if the specified directories are
+        already owned by the right user and group, files and directories below of them are left as-is, even if they do
+        not match what is requested.</para>
+
+        <para>These options imply <varname>ReadWritePaths=</varname> for the specified paths. When combined with
+        <varname>RootDirectory=</varname> or <varname>RootImage=</varname> these paths always reside on the host and
+        are mounted from there into the unit's file system namespace. If <varname>DynamicUser=</varname> is used in
+        conjunction with <varname>StateDirectory=</varname>, <varname>CacheDirectory=</varname> and
+        <varname>LogsDirectory=</varname>, the behaviour of these options is slightly altered: the directories are
+        created below <filename>/var/lib/private</filename>, <filename>/var/cache/private</filename>,
+        <filename>/var/log/private</filename>, respectively, which are host directories made inaccessible to
+        unprivileged users, which ensures that access to these directories cannot be gained through dynamic user ID
+        recycling. Symbolic links are created to hide this difference in behaviour. Both from perspective of the host
+        and from inside the unit, the relevant directories hence always appear directly below
+        <filename>/var/lib</filename>, <filename>/var/cache</filename> and
+        <filename>/var/log</filename>.</para></listitem>
```

Comment:
Currently, this logic is also applied to `RuntimeDirectory=`. I think we should merge the entry for `RuntimeDirectory=` and one for `StateDirectory=`.

Reply:
done